crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -0,0 +1,948 @@
+            // Operation::fdiv
+
+            // Operation::sgt
+            // Operation::sge
+            // Operation::slt
+            // Operation::sle
+
+            // Operation::gt
+            // Operation::lte
+            // Operation::gte
+            // Operation::cast =
+            // Operation::trunc
+            // jumps, ass, nop => do not evaluate
+            // Operation::ass => todo!(),
+            // Operation::nop => todo!(),
+        }
+        NodeEval::Idx(self.idx)
+    }
+
+    // Simplifies trivial Phi instructions by returning:
+    // None, if the instruction is unreachable or in the root block and can be safely deleted
+    // Some(id), if the instruction can be replaced by the node id
+    // Some(ins_id), if the instruction is not trivial
+    pub fn simplify_phi(
+        ins_id: arena::Index,
+        phi_arguments: &[(arena::Index, arena::Index)],
+    ) -> Option<arena::Index> {
+        let mut same = None;
+        for op in phi_arguments {
+            if Some(op.0) == same || op.0 == ins_id {
+                continue;
+            }
+            if same.is_some() {
+                //no simplification
+                return Some(ins_id);
+            }
+
+            same = Some(op.0);
+        }
+        //if same.is_none()  => unreachable phi or in root block, can be replaced by ins.lhs (i.e the root) then.
+        same
+    }
+
+    pub fn standard_form(&mut self) {
+        match self.operator {
+            //convert greater into less
+            Operation::ugt => {
+                std::mem::swap(&mut self.rhs, &mut self.lhs);
+                self.operator = Operation::ult
+            }
+            Operation::uge => {
+                std::mem::swap(&mut self.rhs, &mut self.lhs);
+                self.operator = Operation::ule
+            }
+            Operation::sgt => {
+                std::mem::swap(&mut self.rhs, &mut self.lhs);
+                self.operator = Operation::slt
+            }
+            Operation::sge => {
+                std::mem::swap(&mut self.rhs, &mut self.lhs);
+                self.operator = Operation::sle
+            }
+            Operation::gt => {
+                std::mem::swap(&mut self.rhs, &mut self.lhs);
+                self.operator = Operation::lt
+            }
+            Operation::gte => {
+                std::mem::swap(&mut self.rhs, &mut self.lhs);
+                self.operator = Operation::lte
+            }
+            //TODO replace a<b with a<=b+1, but beware of edge cases!
+            Operation::eq_gate => {
+                if self.rhs == self.lhs {
+                    self.rhs = self.idx;
+                    self.is_deleted = true;
+                }
+            }
+            _ => (),
+        }
+        if is_commutative(self.operator) && self.rhs < self.lhs {
+            std::mem::swap(&mut self.rhs, &mut self.lhs);
+        }
+    }
+}
+
+//adpated from LLVM IR
+#[allow(non_camel_case_types)]
```

Comment:
Lets remove this and use PascalCase

Reply:
I would prefer to keep these opcodes close to llvm, they are not object, struct nor function so I think it is ok for them not to use PascalCase.
For instance, the Add() function for arithmetic gates is a function and therefore follows the PascalCase, then having add opcode in PascalCase Add would be more confusing for me.

Reply:
I don't think we should follow the casing of LLVM if it's not inline with the Rust convention, because its important to have consistency throughout the codebase and strive for the code to look like other Rust codebases. 

Functions in Rust follow snake_case, so I think the `Add()` method would be `add()`.
If it is the case that we have a function named just `add`,  this is also indicative that it probably should be a method on a struct, a trait implementation, or it should be named to be more descriptive, like `add_arithmetic_linear`, for example.

I have seen some cases for not using PascalCase/UpperCamelCase for conveying mathematical meaning, but even then it could be argued that it's better to stick with the conventional case, because no convention is better than the other, its just about consistency and making everything look the same across the ecosystem.

For reference, here is a link to some of the naming conventions: https://rust-lang.github.io/api-guidelines/naming.html